### PR TITLE
feature/implement-google-OA

### LINKED ISSUE #9: feature/implement-google-OA
practicing implementation of the google online auth (which allows a user to login with his google account) 

this feature probably won't be kept as is but could be implemented later on in a different manner.

### PR DESCRIPTION
**Pull request by Transurgeon**
Hey guys,

I have completed this issue

users are now able to sign up using their google account (with a popup)

also it displays the user's name under the hello button in the main page

Resolves #9

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     "framer-motion": "^6.5.1",
     "next": "^12.3.1",
     "react": "18.2.0",
-    "react-dom": "18.2.0"
+    "react-dom": "18.2.0",
```
Comment (Transurgeon) [inline package.json:24]: I added react-firebase-hooks to the npm packages,
so we will have to reinstall npm

File: pages/index.tsx [hunk 2]
```diff
@@ -25,11 +43,23 @@ export default function Home({ welcomeUser }: { welcomeUser: User }) {
         animate={{ opacity: 1, x: 0 }}
         exit={{ opacity: 0, x: 1000 }}
       >
+        <Head>
+          <title>MoreGlean</title>
```
Comment (Transurgeon) [inline pages/index.tsx]: added title and head

File: pages/index.tsx [hunk 3]
```diff
@@ -11,7 +16,20 @@ export async function getServerSideProps() {
   return { props: { welcomeUser } }
 }
 
-export default function Home({ welcomeUser }: { welcomeUser: User }) {
+export default function Home(
+  this: any,
```
Comment (Transurgeon) [inline pages/index.tsx]: here is the implementation of the authentication

**Review by Transurgeon — COMMENTED** [review]
please approve or propose any changes

(favicon still doesn't work for some reason, we can figure it out later I think)

**Review by diegodofe — CHANGES_REQUESTED** [review]
Getting a firebase error when I clicked the sign in button

Screen Shot 2022-11-05 at 4 25 35 PM

File: pages/index.tsx [hunk 4]
```diff
@@ -1,23 +1,53 @@
...
+          flexDirection: 'column',
+          alignItems: 'center',
+          gap: 32,
+        }}
+      >
+        <h1 style={{ fontSize: 'clamp(28px, 6vw, 46px)' }}>
+          Welcome to MoreGlean!
+        </h1>
+
+        <Button type='primary' onClick={handleLogin} disabled>
```
Comment (diegodofe) [inline pages/index.tsx:43]: Thanks for disabling this for now

**Review by diegodofe — APPROVED** [review]

**Pull request merged**